sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/ThresholdBasedAvailabilityStrategy.java

```diff
@@ -0,0 +1,92 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.cosmos;
+
+import java.time.Duration;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+
+/**
+ * The type Threshold based retry availability strategy.
+ */
+public class ThresholdBasedAvailabilityStrategy extends AvailabilityStrategy{
+    private static final Duration DEFAULT_THRESHOLD_IN_MILLISECONDS = Duration.ofMillis(500);
```

Comment:
Since this is duration, we don't need the milliseconds suffix. 
```suggestion
    private static final Duration DEFAULT_THRESHOLD = Duration.ofMillis(500);
```